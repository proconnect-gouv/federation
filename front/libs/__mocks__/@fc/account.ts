import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
