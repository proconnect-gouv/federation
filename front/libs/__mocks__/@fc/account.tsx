import React from 'react';

export const AccountProvider = jest.fn(({ children }) => (
  <div>
    <div>AccountProvider</div>
    <div>{children}</div>
  </div>
));

export const AccountContext = React.createContext(expect.any(Object));
