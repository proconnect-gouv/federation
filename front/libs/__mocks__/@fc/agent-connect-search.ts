import * as React from 'react';

export const AgentConnectSearchContext = React.createContext({
  hasSearched: false,
  payload: {
    csrfToken: 'csrf-token-mock',
    identityProviders: [],
    isLoaded: false,
    ministries: [],
    serviceProviderName: undefined,
  },
  searchResults: [],
  setSearchTerm: () => {},
});

export const useSearchResults = jest.fn(() => ({
  searchResults: [
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.1',
          uid: 'mock-idp-uid-1.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-1.2',
          uid: 'mock-idp-uid-1.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-1',
        identityProviders: ['mock-idp-uid-1'],
        name: 'mock-ministry-name-1',
        sort: 1,
      },
    },
    {
      identityProviders: [
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.1',
          uid: 'mock-idp-uid-2.1',
        },
        {
          active: true,
          display: true,
          name: 'mock-idp-name-2.2',
          uid: 'mock-idp-uid-2.2',
        },
      ],
      ministry: {
        id: 'mock-ministry-id-2',
        identityProviders: ['mock-idp-uid-2'],
        name: 'mock-ministry-name-2',
        sort: 0,
      },
    },
  ],
  showNoResults: false,
  showResults: true,
}));

export const useOnSearch = jest.fn(() => jest.fn());
