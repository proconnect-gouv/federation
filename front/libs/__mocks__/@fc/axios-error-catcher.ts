import React from 'react';

export const AxiosErrorCatcherProvider = jest.fn();

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));
