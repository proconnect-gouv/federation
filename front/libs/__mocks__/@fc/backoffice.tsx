export const FieldCheckboxComponent = jest.fn(() => <div>FieldCheckboxComponent</div>);

export const FieldSwitchComponent = jest.fn(() => <div>FieldSwitchComponent</div>);

export const ButtonSimpleComponent = jest.fn(() => <div>ButtonSimpleComponent</div>);
