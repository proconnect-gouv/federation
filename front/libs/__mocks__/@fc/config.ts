export const ConfigService = {
  get: jest.fn(),
  initialize: jest.fn(),
};
