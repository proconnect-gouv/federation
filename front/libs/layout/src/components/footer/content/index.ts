/* istanbul ignore file */

// declarative file
export * from './layout-footer.content';
