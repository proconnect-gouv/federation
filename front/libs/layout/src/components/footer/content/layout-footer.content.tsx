import React from 'react';

import { ConfigService } from '@fc/config';

import { Options } from '../../../enums';
import type { LayoutConfig } from '../../../interfaces';

interface LayoutFooterContentComponentProps {
  showIcon?: boolean;
}

export const LayoutFooterContentComponent = React.memo(
  ({ showIcon = false }: LayoutFooterContentComponentProps) => {
    const config = ConfigService.get<LayoutConfig>(Options.CONFIG_NAME);
    const { description, links } = config.footer;

    const target = showIcon ? '_blank' : undefined;
    const rel = showIcon ? 'noreferrer' : undefined;
    return (
      <div className="fr-footer__content">
        {description && <p className="fr-footer__content-desc">{description}</p>}
        {links && (
          <ul className="fr-footer__content-list">
            {links.map(({ href, label, title }, index) => {
              const uniqKey = `layout-footer-content-links::${index}`;
              return (
                <li key={uniqKey} className="fr-footer__content-item">
                  <a
                    className="fr-footer__content-link"
                    href={href}
                    rel={rel}
                    target={target}
                    title={title}>
                    {label}
                  </a>
                </li>
              );
            })}
          </ul>
        )}
      </div>
    );
  },
);

LayoutFooterContentComponent.displayName = 'LayoutFooterContentComponent';
