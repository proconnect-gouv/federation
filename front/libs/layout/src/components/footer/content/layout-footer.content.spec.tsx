import { render } from '@testing-library/react';

import { ConfigService } from '@fc/config';

import { LayoutFooterContentComponent } from './layout-footer.content';

describe('LayoutFooterContentComponent', () => {
  // given
  const navigationMock = [
    {
      href: 'any-href-mock-1',
      label: 'any-label-mock-1',
      title: 'any-title-mock-1',
    },
    {
      href: 'any-href-mock-2',
      label: 'any-label-mock-2',
      title: 'any-title-mock-2',
    },
  ];

  beforeEach(() => {
    // given
    jest.mocked(ConfigService.get).mockReturnValue({
      footer: {
        description: undefined,
        links: navigationMock,
      },
    });
  });

  it('should call ConfigService.get with Layout config name', () => {
    // when
    render(<LayoutFooterContentComponent />);

    // then
    expect(ConfigService.get).toHaveBeenCalledOnce();
    expect(ConfigService.get).toHaveBeenCalledWith('Layout');
  });

  it('should match the snapshot when description and links are defined', async () => {
    // given
    jest.mocked(ConfigService.get).mockReturnValue({
      footer: {
        description: 'any description mock',
        links: navigationMock,
      },
    });

    // when
    const { container, findAllByRole, getByTitle } = render(<LayoutFooterContentComponent />);
    const elementLinks = await findAllByRole('link');
    const elementLink1 = getByTitle('any-title-mock-1');
    const elementLink2 = getByTitle('any-title-mock-2');

    // then
    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('any description mock');
    expect(elementLinks).toHaveLength(2);
    expect(elementLink1).toBeInTheDocument();
    expect(elementLink1).toHaveAttribute('href', 'any-href-mock-1');
    expect(elementLink1).toHaveAttribute('title', 'any-title-mock-1');
    expect(elementLink1).not.toHaveAttribute('rel');
    expect(elementLink1).not.toHaveAttribute('target');
    expect(elementLink1).toHaveTextContent('any-label-mock-1');
    expect(elementLink2).toBeInTheDocument();
    expect(elementLink2).toHaveAttribute('href', 'any-href-mock-2');
    expect(elementLink2).toHaveAttribute('title', 'any-title-mock-2');
    expect(elementLink2).not.toHaveAttribute('rel');
    expect(elementLink2).not.toHaveAttribute('target');
    expect(elementLink2).toHaveTextContent('any-label-mock-2');
  });

  it('should match the snapshot when showIcon is true', async () => {
    // given
    jest.mocked(ConfigService.get).mockReturnValue({
      footer: {
        description: 'any description mock',
        links: navigationMock,
      },
    });

    // when
    const { container, findAllByRole, getByTitle } = render(
      <LayoutFooterContentComponent showIcon />,
    );
    const elementLinks = await findAllByRole('link');
    const elementLink1 = getByTitle('any-title-mock-1');
    const elementLink2 = getByTitle('any-title-mock-2');

    // then
    expect(container).toMatchSnapshot();
    expect(container).toHaveTextContent('any description mock');
    expect(elementLinks).toHaveLength(2);
    expect(elementLink1).toBeInTheDocument();
    expect(elementLink1).toHaveAttribute('href', 'any-href-mock-1');
    expect(elementLink1).toHaveAttribute('title', 'any-title-mock-1');
    expect(elementLink1).toHaveAttribute('target', '_blank');
    expect(elementLink1).toHaveAttribute('rel', 'noreferrer');
    expect(elementLink1).toHaveTextContent('any-label-mock-1');
    expect(elementLink2).toBeInTheDocument();
    expect(elementLink2).toHaveAttribute('href', 'any-href-mock-2');
    expect(elementLink2).toHaveAttribute('title', 'any-title-mock-2');
    expect(elementLink2).toHaveAttribute('target', '_blank');
    expect(elementLink2).toHaveAttribute('rel', 'noreferrer');
    expect(elementLink2).toHaveTextContent('any-label-mock-2');
  });
});
