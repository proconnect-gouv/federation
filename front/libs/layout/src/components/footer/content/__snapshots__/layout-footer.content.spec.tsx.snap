// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutFooterContentComponent should match the snapshot when description and links are defined 1`] = `
<div>
  <div
    class="fr-footer__content"
  >
    <p
      class="fr-footer__content-desc"
    >
      any description mock
    </p>
    <ul
      class="fr-footer__content-list"
    >
      <li
        class="fr-footer__content-item"
      >
        <a
          class="fr-footer__content-link"
          href="any-href-mock-1"
          title="any-title-mock-1"
        >
          any-label-mock-1
        </a>
      </li>
      <li
        class="fr-footer__content-item"
      >
        <a
          class="fr-footer__content-link"
          href="any-href-mock-2"
          title="any-title-mock-2"
        >
          any-label-mock-2
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`LayoutFooterContentComponent should match the snapshot when showIcon is true 1`] = `
<div>
  <div
    class="fr-footer__content"
  >
    <p
      class="fr-footer__content-desc"
    >
      any description mock
    </p>
    <ul
      class="fr-footer__content-list"
    >
      <li
        class="fr-footer__content-item"
      >
        <a
          class="fr-footer__content-link"
          href="any-href-mock-1"
          rel="noreferrer"
          target="_blank"
          title="any-title-mock-1"
        >
          any-label-mock-1
        </a>
      </li>
      <li
        class="fr-footer__content-item"
      >
        <a
          class="fr-footer__content-link"
          href="any-href-mock-2"
          rel="noreferrer"
          target="_blank"
          title="any-title-mock-2"
        >
          any-label-mock-2
        </a>
      </li>
    </ul>
  </div>
</div>
`;
